### PR TITLE
feat: update and cleanup solution

### PR DESCRIPTION
**Pull request by alsami**
* Upgrade projects to new csproj-format where possible
* Update packages and unify dotnet-core app version to 3.1 where possible (AspNetCoreExample for <= 3.0 excluded)
* Update packages and remove not required direct references

### CODE REVIEW

**Comment by alsami** [comment]
Still need to wait for some package updates in order to be able to complete that.

**Comment by alsami** [comment]
@alexmg @alistairjevans @tillig the package [Autofac.Extras.EnterpriseLibraryConfigurator](https://github.com/autofac/Autofac.Extras.EnterpriseLibraryConfigurator/) hasn't been updated since 2015. Which makes it impossible to update packages of the project `EnterpriseLibraryExample.MvcApplication`.

I'd delete that, given support for it has been dropped 2015. What do you think?

**Comment by tillig** [comment]
I'm fine dropping the entlib sample.

**Comment by alsami** [comment]
Okay, looks good so far. Got everything updated. Now waiting for the last package to be released (ServiceFabric).
If anyone wants to take a look on the current changes, that would be appreciated.

File: src/AttributeMetadataExample/AttributeMetadataExample.csproj [hunk 1]
```diff
@@ -1,80 +1,20 @@
...
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+    <AssemblyTitle>AttributeMetadataExample</AssemblyTitle>
+    <Company>Microsoft Corporation</Company>
```
Comment (alexmg) [inline src/AttributeMetadataExample/AttributeMetadataExample.csproj:7]: Not sure why the original example had attributes referring to Microsoft but I think we should just remove them.

**Review by alexmg — CHANGES_REQUESTED** [review]

File: src/AttributeMetadataExample/AttributeMetadataExample.csproj (hunk 1 above)
Comment (alsami) [reply src/AttributeMetadataExample/AttributeMetadataExample.csproj]: Removed those everywhere I found them, also in the Assembly-Info.

**Comment by alsami** [comment]
The only thing that is missing is the ServiceFabric sample. I think I could take on this later when the package is done so we can get that merged. What do you think?

**Comment by alexmg** [comment]
The Service Fabric sample can wait for the package release. Splendid work with the updates! 👍

**Pull request merged**